sidecar/internal/reconciler/bucket.go

```diff
@@ -342,37 +343,37 @@ func objectProtocolListFromApiList(apiList []cosiapi.ObjectProtocol) ([]*cosipro
 	}
 
 	if len(errs) > 0 {
-		return nil, fmt.Errorf("failed to parse protocol list: %v", errs)
+		return nil, fmt.Errorf("failed to parse protocol list: %w", errors.Join(errs...))
 	}
 	return out, nil
 }
 
 // validate that the required protocols (if given) are supported by the driver
 func validateDriverSupportsProtocols(driver DriverInfo, required []*cosiproto.ObjectProtocol) error {
-	unsupportedProtos := []string{}
+	unsupportedProtos := []error{}
 
 	for _, proto := range required {
 		if !driver.SupportsProtocol(proto.Type) {
-			unsupportedProtos = append(unsupportedProtos, proto.Type.String())
+			unsupportedProtos = append(unsupportedProtos, fmt.Errorf("%s", proto.Type))
 		}
 	}
 
 	if len(unsupportedProtos) > 0 {
-		return fmt.Errorf("driver %q does not support protocols: %v", driver.name, unsupportedProtos)
+		return fmt.Errorf("driver %q does not support protocols: %w", driver.name, errors.Join(unsupportedProtos...))
```

Comment:
This is just a list of unsupported protocols, not necessarily errors. Let's revert this small set of changes.